cax/tasks/process_hax.py

```diff
@@ -36,16 +36,31 @@ def _process_hax(name, in_location, host, pax_version,
 
     os.makedirs(out_location, exist_ok=True)
 
-    try:
-        print('creating hax minitrees for run', name, pax_version, in_location, out_location)
-        init_hax(in_location, pax_version, out_location)   # may initialize once only
-        hax.minitrees.load_single_dataset(name, ['Corrections', 'Basics', 'Fundamentals',
-                                                 'CorrectedDoubleS1Scatter', 'LargestPeakProperties',
-                                                 'TotalProperties',  'Extended', 'Proximity','LoneSignalsPreS1', 
-                                                 'LoneSignals', 'FlashIdentification'])
-
-    except Exception as exception:
-        raise
+    init_hax(in_location, pax_version, out_location)   # may initialize once only
+
+    print('creating hax minitrees for run', name, pax_version, in_location, out_location)
+
+    TREEMAKERS = ['Corrections', 'Basics', 'Fundamentals',
+                  'CorrectedDoubleS1Scatter', 'LargestPeakProperties',
+                  'TotalProperties', 'Extended', 'Proximity', 'LoneSignalsPreS1',
+                  'LoneSignals', 'FlashIdentification']
+
+    for treemaker in TREEMAKERS:
+
+        nRetries = 5
+
+        while nRetries > 0:
+            try:
+                print('Creating minitree', treemaker)
+                hax.minitrees.load_single_dataset(name, treemaker)
+                nRetries = 0
+
+            except Exception as exception:
+                os.remove("%s/%s_%s.root" % (out_location, name, treemaker))
+                print("Deleting %s/%s_%s.root" % (out_location, name, treemaker))
+                nRetries -= 1
+                if nRetries == 0:
+                    raise
```

Comment:
Rise what?

Reply:
it was here before. it just stops the program and prints the error. should help with detection if there's a bigger problem